python/lsst/base/packages.py

```diff
@@ -74,7 +74,8 @@ def getVersionFromPythonModule(module):
         deps = module.__dependency_versions__
         buildtime = BUILDTIME & set(deps.keys())
         if buildtime:
-            version += " with " + " ".join("%s=%s" % (pkg, deps[pkg]) for pkg in buildtime)
+            version += " with " + " ".join("%s=%s" % (pkg, deps[pkg])
+                                           for pkg in sorted(buildtime))
```

Comment:
Suggest leaving the `for` on the previous line, which makes it clear that the line continues (split after an 'operator').